cilium/CFP-32820-improve-scalability-of-health-probing.md

```diff
@@ -0,0 +1,158 @@
+# CFP-32820: Improve Scalability of Health Probing
+
+**SIG: SIG-Scalability**
+
+**Begin Design Discussion:** 2024-08-12
+
+**Cilium Release:** 1.17
+
+**Authors:** Shreya Jayaraman <shjayaraman@microsoft.com> based on the work of Marcel Zięba <marcel.zieba@isovalent.com>
+
+**Status:** Implementable
+
+## Summary
+
+Cilium currently recommends disabling ICMP and HTTP health probes for large-scale clusters due to their impacts on cpu/mem usage. We propose to dynamically set the probing interval and thereby provide a configurable number of queries per second issued by probers to reduce the effects of ping frequency at scale.
+
+## Motivation
+
+Currently, Cilium uses runs ICMP and HTTP probes at intervals of 60 seconds, which may be an issue in large-scale clusters and is why Cilium currently [recommends that users disable health checking](https://docs.cilium.io/en/stable/operations/performance/scalability/report/) in such clusters.
+
+There is a trade-off between resource consumption (for e.g., cpu/mem usage on the cluster and network bandwidth) and the frequency of updates regarding connectivity status. The method of health probing proposed here intends to allow Cilium users to determine which is more important to them. For instance, a cluster administrator for a large cluster who wants to ensure health checking does not have significant effects on bandwidth or resource use may find it reasonable to have the health probes run less frequently, even if the connectivity data is provided less often. On the other hand, a cluster administrator may want to have connectivity statuses updated more frequently, despite its negative effects. This proposal intends to provide users with a way to find a balance between probe frequency and resource use that reflects their needs.
+
```

Comment:
Does it make sense to add these here?

```suggestion

### User Journeys

There are two conflicting user profiles to testing cluster connectivity which this CFP aims to address:

1. As a cluster administrator, I have a large cluster and want to ensure that the network bandwidth consumed for validating basic connectivity across nodes is well-bounded, even if it means that Cilium can less quickly tell when there is a connectivity breakage on the underlying network.
1. As a cluster administrator, I want to be able to quickly tell whether basic connectivity is OK across nodes, potentially even at the expense of network bandwidth usage within the environment.
```

(Ideally we would hear these directly from users rather than me making them up, but I think that these opinions sound reasonable and they motivate 'why' to pick Option 1 as the solution).

Reply:
Makes sense to me :) I've pulled out the user journey statements from the paragraph to this more explicit form.